src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -1152,4 +1156,117 @@ public void testIsAsyncSession() throws SQLException, InterruptedException {
+    // 4.) After the test is complete, remove the files from the resources folder
+
+    final String privateKeyFileNameEnv = "SNOWFLAKE_PRIVATE_KEY_FILENAME";
+    final String publicKeyFileNameEnv = "SNOWFLAKE_PUBLIC_KEY_FILENAME";
+    final String passphraseEnv = "SNOWFLAKE_TEST_PASSPHRASE";
+
+    String privateKeyFile = System.getenv(privateKeyFileNameEnv);
+    String publicKeyFile = System.getenv(publicKeyFileNameEnv);
+    String passphrase = System.getenv(passphraseEnv);
+
+    assertNotNull(
+        privateKeyFileNameEnv
+            + " environment variable can't be empty. "
+            + "Please provide the filename for your private key located in the resource folder",
+        passphrase);
+
+    assertNotNull(
+        publicKeyFileNameEnv
+            + " environment variable can't be empty. "
+            + "Please provide the filename for your public key located in the resource folder",
+        passphrase);
+
+    assertNotNull(
+        passphraseEnv + " environment variable is required to decrypt private key.", passphrase);
+    Map<String, String> parameters = getConnectionParameters();
+    String testUser = parameters.get("user");
+    Properties properties = new Properties();
+    properties.put("account", parameters.get("account"));
+    properties.put("user", testUser);
+    properties.put("ssl", parameters.get("ssl"));
+    properties.put("port", parameters.get("port"));
+    Connection connection = getConnection();
+    Statement statement = connection.createStatement();
+    statement.execute("use role accountadmin");
+    String pathFile = getFullPathFileInResource(publicKeyFile);
+    String pubKey = new String(Files.readAllBytes(Paths.get(pathFile)));
+    pubKey = pubKey.replace("-----BEGIN PUBLIC KEY-----", "");
+    pubKey = pubKey.replace("-----END PUBLIC KEY-----", "");
+    pubKey = pubKey.replace("\n", "");
+    statement.execute(String.format("alter user %s set rsa_public_key='%s'", testUser, pubKey));
+    connection.close();
+    String privateKeyLocation = getFullPathFileInResource(privateKeyFile);
+    String uri =
+        parameters.get("uri")
+            + "/?private_key_file_pwd="
+            + passphrase
+            + "&private_key_file="
+            + privateKeyLocation;
+    try {
+      connection = DriverManager.getConnection(uri, properties);
+    } catch (SQLException e) {
+      connection.close();
+      return false;
+    }
+
+    // clean up
+    connection = getConnection();
+    statement = connection.createStatement();
+    statement.execute("use role accountadmin");
+    statement.execute(String.format("alter user %s unset rsa_public_key", testUser));
+    connection.close();
+    return true;
+  }
+
+  @Test
+  @Ignore
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testPbes2Support() throws SQLException, IOException {
+    // This test is ignored and won't run until you explicitly uncomment the
```

Comment:
let's move the whole comment how to setup the test to the javadoc above the test method